## 1. The World Bank's international debt data
<p>The aim of this project is to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories.</p>
<p>This tries to find the answers to questions like: </p>
<ul>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>

In [1]:
%load_ext sql
%sql postgresql://postgres:***@localhost:5433/international_debt
%config SqlMagic.autopandas = False
%config SqlMagic.style = '_DEPRECATED_DEFAULT'

In [2]:
%%sql
SELECT *
FROM international_debt
LIMIT 10;

 * postgresql://postgres:***@localhost:5433/international_debt
10 rows affected.


country_name,country_code,indicator_name,indicator_code,debt
Afghanistan,AFG,"Disbursements on external debt, long-term (DIS, current US$)",DT.DIS.DLXF.CD,72894453.700000003
Afghanistan,AFG,"Interest payments on external debt, long-term (INT, current US$)",DT.INT.DLXF.CD,53239440.100000001
Afghanistan,AFG,"PPG, bilateral (AMT, current US$)",DT.AMT.BLAT.CD,61739336.899999999
Afghanistan,AFG,"PPG, bilateral (DIS, current US$)",DT.DIS.BLAT.CD,49114729.399999999
Afghanistan,AFG,"PPG, bilateral (INT, current US$)",DT.INT.BLAT.CD,39903620.100000001
Afghanistan,AFG,"PPG, multilateral (AMT, current US$)",DT.AMT.MLAT.CD,39107845
Afghanistan,AFG,"PPG, multilateral (DIS, current US$)",DT.DIS.MLAT.CD,23779724.300000001
Afghanistan,AFG,"PPG, multilateral (INT, current US$)",DT.INT.MLAT.CD,13335820
Afghanistan,AFG,"PPG, official creditors (AMT, current US$)",DT.AMT.OFFT.CD,100847181.900000006
Afghanistan,AFG,"PPG, official creditors (DIS, current US$)",DT.DIS.OFFT.CD,72894453.700000003


## 2. Finding the number of distinct countries
<p>From the first ten rows, it's clear that the amount of debt owed by <em>Afghanistan</em> in the different debt indicators. But the number of different countries on the table are unknown to viewers. There are repetitions in the country names because a country is most likely to have debt in more than one debt indicator. </p>
<p>Without a count of unique countries, statistical analyses will not be able to perform holistically. So in this section, I extract the number of unique countries present in the table. </p>

In [3]:
%%sql
SELECT 
    COUNT(DISTINCT(country_name)) AS total_distinct_countries
FROM international_debt;

 * postgresql://postgres:***@localhost:5433/international_debt
1 rows affected.


total_distinct_countries
124


## 3. Finding out the distinct debt indicators
<p>There are a total of 124 countries present on the table. In the first section, there is a column called <code>indicator_name</code> that briefly specifies the purpose of taking the debt. Just beside that column, there is another column called <code>indicator_code</code> which symbolizes the category of these debts. Knowing about these various debt indicators will help us to better understand the areas in which a country can possibly be indebted to. </p>

In [4]:
%%sql
SELECT 
    DISTINCT(indicator_code) as distinct_debt_indicators
FROM international_debt
GROUP BY distinct_debt_indicators
ORDER BY distinct_debt_indicators;

 * postgresql://postgres:***@localhost:5433/international_debt
25 rows affected.


distinct_debt_indicators
DT.AMT.BLAT.CD
DT.AMT.DLXF.CD
DT.AMT.DPNG.CD
DT.AMT.MLAT.CD
DT.AMT.OFFT.CD
DT.AMT.PBND.CD
DT.AMT.PCBK.CD
DT.AMT.PROP.CD
DT.AMT.PRVT.CD
DT.DIS.BLAT.CD


## 4. Totaling the amount of debt owed by the countries
<p>The financial debt of a particular country represents its economic state.</p>
<p>The next step is to switch gears from the debt indicators and find out the total amount of debt (in USD) that is owed by the different countries. This will give us a sense of how the overall economy of the entire world is holding up.</p>

In [5]:
%%sql
SELECT 
    ROUND(SUM(debt)/1000000, 2) as total_debt
FROM international_debt;

 * postgresql://postgres:***@localhost:5433/international_debt
1 rows affected.


total_debt
3079734.49


## 5. Country with the highest debt
<p>Next step is to find out the country that owns the highest amount of debt along with the amount. 

In [7]:
%%sql
SELECT 
    country_name, 
    ROUND(SUM(debt)/1000000, 2) as total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC 
LIMIT 1;

 * postgresql://postgres:***@localhost:5433/international_debt
1 rows affected.


country_name,total_debt
China,285793.49


## 6. Average amount of debt across indicators
<p>Next is to find out on an average how much debt a country owes. This will give us a better sense of the distribution of the amount of debt across different indicators.</p>

In [8]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    ROUND(AVG(debt)/1000000,2) as average_debt,
    indicator_name
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10;

 * postgresql://postgres:***@localhost:5433/international_debt
10 rows affected.


debt_indicator,average_debt,indicator_name
DT.AMT.DLXF.CD,5904.87,"Principal repayments on external debt, long-term (AMT, current US$)"
DT.AMT.DPNG.CD,5161.19,"Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)"
DT.DIS.DLXF.CD,2152.04,"Disbursements on external debt, long-term (DIS, current US$)"
DT.DIS.OFFT.CD,1958.98,"PPG, official creditors (DIS, current US$)"
DT.AMT.PRVT.CD,1803.69,"PPG, private creditors (AMT, current US$)"
DT.INT.DLXF.CD,1644.02,"Interest payments on external debt, long-term (INT, current US$)"
DT.DIS.BLAT.CD,1223.14,"PPG, bilateral (DIS, current US$)"
DT.INT.DPNG.CD,1220.41,"Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)"
DT.AMT.OFFT.CD,1191.19,"PPG, official creditors (AMT, current US$)"
DT.AMT.PBND.CD,1082.62,"PPG, bonds (AMT, current US$)"


## 7. The highest amount of principal repayments
<p>We can see that the indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital. </p>
<p>Interestingly, the above findings show that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.</p>
<p>I'll investigate this a bit more so as to find out which country owes the highest amount of debt in the category of long term debts (<code>DT.AMT.DLXF.CD</code>). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically. </p>

In [9]:
%%sql
SELECT 
    country_name, 
    ROUND(AVG(debt)/1000000,2) as average_debt,
    indicator_name,
    indicator_code
FROM international_debt
GROUP BY country_name, indicator_name, indicator_code
HAVING indicator_code = 'DT.AMT.DLXF.CD'
ORDER BY average_debt DESC
LIMIT 10;

 * postgresql://postgres:***@localhost:5433/international_debt
10 rows affected.


country_name,average_debt,indicator_name,indicator_code
China,96218.62,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
Brazil,90041.84,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
Russian Federation,66589.76,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
Turkey,51555.03,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
South Asia,48756.30,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
India,31923.51,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
Indonesia,30916.11,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
Kazakhstan,27482.09,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
Mexico,25218.50,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD
Least developed countries: UN classification,25197.03,"Principal repayments on external debt, long-term (AMT, current US$)",DT.AMT.DLXF.CD


## 8. The most common debt indicator
<p>It's clear to see that China has the highest amount of debt in the long-term debt (<code>DT.AMT.DLXF.CD</code>) category. </p>
<p>We saw that long-term debt is the topmost category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? </p>

In [10]:
%%sql
SELECT 
    indicator_name,
    COUNT(indicator_code) as indicator_count
FROM international_debt
GROUP BY indicator_code, indicator_name
ORDER BY indicator_count DESC
LIMIT 20;

 * postgresql://postgres:***@localhost:5433/international_debt
20 rows affected.


indicator_name,indicator_count
"PPG, multilateral (AMT, current US$)",124
"PPG, multilateral (INT, current US$)",124
"PPG, official creditors (INT, current US$)",124
"Principal repayments on external debt, long-term (AMT, current US$)",124
"PPG, official creditors (AMT, current US$)",124
"Interest payments on external debt, long-term (INT, current US$)",124
"Disbursements on external debt, long-term (DIS, current US$)",123
"PPG, official creditors (DIS, current US$)",122
"PPG, bilateral (INT, current US$)",122
"PPG, bilateral (AMT, current US$)",122


## 9. Other viable debt issues and conclusion
<p>There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator <code>DT.AMT.DLXF.CD</code> is also there in the list 
=> All these countries are suffering from a common economic issue. </p>
<p>Now change tracks from <code>debt_indicator</code> and focus on the amount of debt again. Then find out the maximum amount of debt across the indicators along with the respective country names; which will help to identify the other plausible economic issues a country might be going through. Eventually, the debt indicators in which a country owes its highest debt maybe found. </p>
<p>In this notebook, I looked at debt owed by countries across the globe. I extracted a few summary statistics from the data and unraveled some interesting facts and figures. I also validated our findings to make sure the investigations are correct.</p>

In [11]:
%%sql
SELECT
    country_name,
    indicator_name,
    MAX(debt) as maximum_debt
FROM international_debt
GROUP BY country_name, indicator_name
ORDER BY maximum_debt DESC
LIMIT 20;

 * postgresql://postgres:***@localhost:5433/international_debt
20 rows affected.


country_name,indicator_name,maximum_debt
China,"Principal repayments on external debt, long-term (AMT, current US$)",96218620835.699996948
Brazil,"Principal repayments on external debt, long-term (AMT, current US$)",90041840304.100006104
China,"Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)",72392986213.800003052
Russian Federation,"Principal repayments on external debt, long-term (AMT, current US$)",66589761833.5
Turkey,"Principal repayments on external debt, long-term (AMT, current US$)",51555031005.800003052
South Asia,"Principal repayments on external debt, long-term (AMT, current US$)",48756295898.199996948
Brazil,"PPG, private creditors (AMT, current US$)",43598697498.599998474
Russian Federation,"Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)",42800154974.900001526
Brazil,"Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)",41831444053.300003052
Least developed countries: UN classification,"Disbursements on external debt, long-term (DIS, current US$)",40160766261.599998474
